test/quoteSpec.js

```diff
@@ -0,0 +1,240 @@
+'use strict'
+
+const chai = require('chai')
+const moment = require('moment')
+const assert = chai.assert
+const ILQP = require('..').ILQP
+const Packet = require('..').Packet
+const MockPlugin = require('./mocks/mockPlugin')
+const expect = chai.expect
+const chaiAsPromised = require('chai-as-promised')
+chai.use(chaiAsPromised)
+
+describe('ILQP', function () {
+  beforeEach(function () {
+    this.plugin = new MockPlugin()
+    // quote response
+    this.id = '85e04e5c-2357-4033-ac5b-251ce97faf33'
+    this.response = {
+      data: {
+        id: this.id,
+        method: 'quote_response',
+        data: {
+          source_amount: '1',
+          destination_amount: '1',
+          source_connector_account: 'test.example.connie',
+          source_expiry_duration: '0'
+        }
+      }
+    }
+  })
+
+  describe('quote', function () {
+    beforeEach(function () {
+      this.params = {
+        sourceAddress: 'test.example.alice',
+        destinationAddress: 'test.local.bob',
+        sourceAmount: '1',
+        sourceExpiryDuration: '0',
+        destinationExpiryDuration: '0',
+        connectors: [ 'test.example.connie' ]
+      }
+      this.result = {
+        sourceAmount: '1',
+        destinationAmount: '1',
+        connectorAccount: 'test.example.connie',
+        sourceExpiryDuration: '0'
+      }
+
+      this.plugin.sendMessage = (msg) => {
+        this.response.data.id = msg.data.id
+        this.plugin.emit('incoming_message', this.response)
+        return Promise.resolve(null)
+      }
+    })
+
+    it('should return a quote from the connector', function * () {
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should default to getInfo\'s connectors', function * () {
+      // remove manually provided connectors
+      delete this.params.connectors
+
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
```

Comment:
I delete the connectors getting passed it manually, so it has to get them from the getInfo function. I have another test that tells getInfo to not return anything and then also doesn't pass in connectors, to make sure that fails.